### PR TITLE
[snmp-memory] use chunk instead of big list to avoid memory corruption in low memory device

### PR DESCRIPTION
**Pull request by lipxu**
### Description of PR

Summary:
Fixes # (issue)
24450772

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
In a low memory device, the case would be failed due to no enough memory for the big list.

#### How did you do it?
use chunk instead of big list to avoid memory corrption.

#### How did you verify/test it?
Run the memory.py

#### Any platform specific information?
Platform with low memory

### CODE REVIEW

File: tests/snmp/memory.py [hunk 1]
```diff
@@ -1,7 +1,38 @@
...
+    command = "grep MemFree /proc/meminfo | awk '{print $2}'"
+    output = subprocess.check_output(command, shell=True)
+    free_memory = int(output.strip())
+    return free_memory
+
+
+free_memory = get_free_memory()
+total_chars = 512000000
+
+if (free_memory * 1024) > int(total_chars * 130 / 100):
```
Comment (SuvarnaMeenakshi) [inline tests/snmp/memory.py:17]: Can you elaborate on why you are checking against (total_chars * 130 / 100)?

File: tests/snmp/memory.py [hunk 2]
```diff
@@ -1,7 +1,38 @@
...
+    load = []
+    count = 1000
+    for i in range(0, count):
+        load.append([' ' * int(total_chars / count)])
```
Comment (SuvarnaMeenakshi) [inline tests/snmp/memory.py:21]: printing the load array is being done only in else block, what is the reason for that?

File: tests/snmp/memory.py (hunk 1 above)
Comment (lipxu) [reply tests/snmp/memory.py:17]: Thanks for reviewing @SuvarnaMeenakshi , checking the free memory here is for avoiding memory corruption.
If free memory is smaller than 130% percent of total_chars, then use chunk memory instead.
130% here is only for reserving some buffer. thanks

File: tests/snmp/memory.py (hunk 2 above)
Comment (lipxu) [reply tests/snmp/memory.py:21]: The code in if block is the latest code in github, I try not to affect or modify the original code.
the code was modified by the PR https://github.com/sonic-net/sonic-mgmt/pull/9074.
If you think the printing here is necessary, I can add it back, thanks

**Review by SuvarnaMeenakshi — APPROVED** [review]
lgtm.
Will need this fix https://github.com/sonic-net/sonic-mgmt/pull/9176 for this test case to pass.

**Comment by lipxu** [comment]
merged to 202305 
https://github.com/sonic-net/sonic-mgmt/pull/10072

**Pull request merged**